### PR TITLE
Allow to specify confidence thresholds and output path in classify command

### LINKED ISSUE #536: Add the possibility to select confidence thresholds in the classifiy command

### LINKED ISSUE #540: Save classification payload as JSON file
In order to publish classifications as Taskcluster artifacts, the `classify` commands needs an option to output the classify and regressions details as a json file.

1. add a new CLI option `--output` to the `classify` command, requiring a writable path
2. add an option to `classify` to be able to retrieve the results of `classify_regressions` on the output
3. in the `classify` script, serialize the full output as https://gist.github.com/La0/d45602fccc84368d0cb30ab8f0454925#file-classification-json

### PR DESCRIPTION
**Pull request by EvaBardou**
Based on #538 

Closes #536
Closes #540

### CODE REVIEW

File: mozci/console/commands/push.py [hunk 1]
```diff
@@ -30,16 +32,80 @@ class ClassifyCommand(Command):
...
+        except ValueError:
+            self.line("<error>Provided --high-confidence should be a float.</error>")
+            return
+
+        output = self.option("output")
+        if output and not os.path.isdir(output):
+            self.line(
+                "<error>Provided --output should be a valid path towards a writable directory.</error>"
+            )
+            return
```
Comment (marco-c) [inline mozci/console/commands/push.py:57]: Nit: you could create it in case it doesn't exist

File: mozci/push.py [hunk 2]
```diff
@@ -1085,8 +1085,11 @@ def _map_failing_tasks(groups):
         )
 
     def classify(
-        self, confidence_medium: float = 0.8, confidence_high: float = 0.9
-    ) -> PushStatus:
+        self,
+        confidence_medium: float = 0.8,
+        confidence_high: float = 0.9,
+        output_regressions: bool = False,
```
Comment (marco-c) [inline mozci/push.py:1091]: I'd remove this and always return it, so we simplify the code

File: mozci/console/commands/push.py [hunk 3]
```diff
@@ -30,16 +32,78 @@ class ClassifyCommand(Command):
...
+                        group: [
+                            {"task_id": task.id, "label": task.label}
+                            for task in failing_tasks
+                        ]
+                        for group, failing_tasks in regressions.unknown.items()
+                    },
+                },
+            }
+
+            filename = f"{output}/classify_output.json"
```
Comment (marco-c) [inline mozci/console/commands/push.py:99]: Could you put the branch and revision in the file name? This way when we classify multiple pushes they won't overwrite each other.

Comment (EvaBardou) [reply mozci/console/commands/push.py:99]: Sure

**Review by marco-c — APPROVED** [review]

**Pull request merged**